frontend/src/components/Election/Voting/GenericBallotView.js

```diff
@@ -0,0 +1,221 @@
+          <Grid item xs={6}>
+            <img src={candidate.photo_filename} style={{ width: 200, height: 200 }} />
+          </Grid>}
+          </Grid>
+          
+          <Grid container style={{ backgroundColor: rowColor }}>
+          <Grid item xs={1}>
+          </Grid>
+          <Grid item xs={8}>
+            <Typography align='left' s component="p" style={{whiteSpace: 'pre-line'}}>
+              {candidate.bio}
+            </Typography>
+          </Grid>
+        </Grid>
+        </>
+      }
+
+    </>
+  )
+};
+
+// Represents the list of rows corresponding to the list of candidates
+const Rows = ({ candidates, scores, onClick, columns }) =>
+  candidates.map((row, n) => (
+    <>
+      <Row
+        rowIndex={n}
+        key={`starRow${n}`}
+        candidate={row}
+        party={row.party}
+        score={scores[n]}
+        onClick={(score) => onClick(n, score)}
+        columns={columns}
+      />
+      <Divider style={{ width: '100%' }} />
+    </>
+  ));
+
+// Represents the list of column headings for all possible scores
+const ColumnHeadings = ({columns}) => (
+  <>
+    <Grid container alignItems="stretch">
+      <Grid item xs={5}></Grid>
+      <ScoreColumnHeadings columns={columns}/>
+    </Grid>
+  </>
+);
+
+const ScoreColumnHeadings = ({columns}) =>
+  columns.map((columnValue, n) => (
+    <Grid item xs={1}>
+      <Typography align='center' variant="h6" component="h6">
+        {columns.length == 1? ' ' : columnValue}
+      </Typography>
+    </Grid>
+  ));
+
+
+
+// Renders a complete RCV ballot for a single race
+export default function GenericBallotView({
+  race,
+  candidates,
+  scores,
+  onClick,
+  columns,
+  instructions
+}) {
+  return (
+      <Box border={2} sx={{ mt: 5, ml: 0, mr: 0, width: '100%' }}>
+        <Grid container alignItems="center" justify="center" direction="column">
+
+          <Grid item style={{ padding: '0.8cm 0cm 0cm 0cm' }}>
+            <Typography align='center' gutterBottom variant="h2" component="h6">
+              {race.title}
+            </Typography>
+          </Grid>
+          <Grid item>
+            <Typography align='center' gutterBottom variant="h6" component="h6" style={{whiteSpace: 'pre-line'}}>
+              {race.description}
+            </Typography>
+          </Grid>
+
+          <Grid item xs={8} style={{ padding: '0cm 0cm 1cm 0cm' }}>
+            {instructions}
+          </Grid>
+
+          <ColumnHeadings columns={columns}/>
+          <Divider style={{ width: '100%' }} />
+          <Rows candidates={candidates} scores={scores} onClick={onClick} columns={columns}/>
+
+          <Grid item xs={10} style={{ padding: '0.4cm 0cm' }}>
+            {race.num_winners == 1 && race.voting_method == 'STAR' &&
+              <Typography align='center' component="p">
+                This election uses STAR Voting and will elect 1 winner. In STAR Voting the two highest scoring candidates are finalists and the finalist preferred by more voters wins.
```

Comment:
We should move this to StarBallotView and add an input for the tabulation explanation to generic ballot view.

Reply:
Sounds good, I've got a fix for this ready in the next PR